FluentUI/src/main/java/com/microsoft/fluentui/contextualcommandbar/ContextualCommandBar.kt

```diff
@@ -0,0 +1,164 @@
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.util.AttributeSet
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import android.widget.FrameLayout
+import android.widget.ImageView
+import com.microsoft.fluentui.R
+import com.microsoft.fluentui.util.isVisible
+
+class ContextualCommandBar @JvmOverloads constructor(
+        context: Context,
+        attrs: AttributeSet? = null,
+        defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+    private var dismissButtonContainer: ViewGroup? = null
+    private var commandItemAdapter: CommandItemAdapter? = null
+    private var commandItemRecyclerView: RecyclerView? = null
+
+    var dismissCommandItem: DismissCommandItem? = null
+        set(value) {
+            field = value
+
+            updateDismissButton()
+        }
+
+    fun setItemOnClickListener(listener: CommandItemAdapter.OnItemClickListener) {
+        commandItemAdapter?.itemClickListener = listener
+    }
+
+    private fun updateDismissButton() {
+        dismissCommandItem ?: return
+        val icon = dismissCommandItem!!.getIcon()
+        if (icon == 0) {
+            return
+        }
+        val dismissItemVisible = dismissCommandItem!!.visible
+        val dismissItemGravity = dismissCommandItem!!.gravity
+
+        var dismissButton: ImageView? = null
+        var dismissButtonDivider: View? = null
+
+        if (dismissButtonContainer == null) {
+            dismissButtonContainer = LayoutInflater.from(context)
+                    .inflate(R.layout.view_dismiss_command_item, null) as ViewGroup
+            addView(dismissButtonContainer)
+            dismissButton = dismissButtonContainer!!.findViewById(R.id.dismiss_command_item_button)
+            dismissButtonDivider = dismissButtonContainer!!.findViewById(R.id.dismiss_command_item_divider)
+        }
+        (dismissButtonContainer!!.layoutParams as LayoutParams).apply {
+            height = MATCH_PARENT
+            width = WRAP_CONTENT
+
+            gravity = when (dismissCommandItem!!.gravity) {
+                DismissItemGravity.START -> Gravity.START
+                DismissItemGravity.END -> Gravity.END
+            }
+        }
+        if (dismissItemGravity == DismissItemGravity.START) {
+            dismissButtonDivider!!.setBackgroundResource(
+                    R.drawable.contextual_command_bar_dismiss_button_divider_start_background
+            )
+            dismissButtonContainer!!.removeAllViews()
+            dismissButtonContainer!!.addView(dismissButton)
+            dismissButtonContainer!!.addView(dismissButtonDivider)
+        } else if (dismissItemGravity == DismissItemGravity.END) {
+            dismissButtonDivider!!.setBackgroundResource(
```

Comment:
JFYI, I am seeing an exception thrown here when I have the Dismiss button displayed at the end and the buttons state updates, kotlin.KotlinNullPointerException, you may be able to repro in the Demo app